src/PowerShellEditorServices.Host/LanguageServer.cs

```diff
@@ -3,98 +3,76 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 //
 
-using DebugAdapterMessages = Microsoft.PowerShell.EditorServices.Protocol.DebugAdapter;
 using Microsoft.PowerShell.EditorServices.Protocol.LanguageServer;
 using Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol;
+using Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol.Channel;
 using Microsoft.PowerShell.EditorServices.Utility;
 using Nito.AsyncEx;
 using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
 using System.Management.Automation;
 using System.Management.Automation.Language;
 using System.Text.RegularExpressions;
 using System.Threading;
 using System.Threading.Tasks;
-using System.IO;
+using DebugAdapterMessages = Microsoft.PowerShell.EditorServices.Protocol.DebugAdapter;
 
 namespace Microsoft.PowerShell.EditorServices.Host
 {
-    internal class LanguageServer : IMessageProcessor
+    internal class LanguageServer : LanguageServerBase 
```

Comment:
Yep, need to add a second constructor that takes a ChannelBase.  The reason why I didn't make this class public yet is that it's only meant to be used in the host process for now.  The way I generally see this working is that other channel implementations would be added to the Protocol library and then exposed through command line flags in the host process.

However, maybe I should just move the server implementations to the Protocol project for maximum flexibility.  Thoughts?